packages/flutter_tools/lib/src/build_system/depfile.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../base/file_system.dart';
+import '../base/platform.dart';
+import '../globals.dart';
+
+/// A class for representing depfile formats.
+class Depfile {
+  /// Create a [Depfile] from a list of [input] files and [output] files.
+  const Depfile(this.inputs, this.outputs);
+
+  /// Parse the depfile contents from [file].
+  ///
+  /// If the syntax is invalid, returns an empty [Depfile].
+  factory Depfile.parse(File file) {
+    final String contents = file.readAsStringSync();
+    final List<String> colonSeparated = contents.split(': ');
+    if (colonSeparated.length != 2) {
+      printError('Invalid depfile: ${file.path}');
+      return const Depfile(<File>[], <File>[]);
```

Comment:
In which instance returning an empty `Depfile` makes the build successful later?

Reply:
In the case where (for some reason, human intervention) an invalid depfile is written or edited - we don't force the user to `flutter clean`, instead we allow a new one be written which hopefully doesn't have the same issue